tools/benchmark/src/mocha/customOutputRunner.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Test } from "mocha";
+
+import type {
+	BenchmarkDescription,
+	HookArguments,
+	MochaExclusiveOptions,
+	Titled,
+} from "../Configuration";
+import type { BenchmarkData } from "../ResultTypes";
+import { timer } from "../timer";
+
+/**
+ * Options to configure a benchmark that reports custom measurements.
+ *
+ * @public
+ */
+export interface CustomBenchmarkOptions
+	extends Titled,
+		BenchmarkDescription,
+		HookArguments,
```

Comment:
On of the really awesome things about this new custom benchmark setup, is since the run function gets to control what is measured, and what is not, there is no need to support these extra hook arguments. Instead the caller can just do whatever is needed at the beginning and end of run.

This should make custom benchmarks much less annoying that our other benchmarks since passing data from setup into the measured part won't be a problem for users anymore.

```suggestion
```

Reply:
Yeah I think Craig is right. I originally thought having before and after on the custom benchmarks made sense, but specifically for benchmark tests (as compared to regular mocha unit tests) one of the main uses for them seems to be just splitting the bits that we don't want to count towards the runtime measurement of a given benchmark... which is not the main point of the custom benchmarks. If those report a slightly higher runtime, I don't think that's necessarily bad or wrong. We'll still get the idea of how long it took to run the whole custom benchmark function.

The code bit that I suggested when we added HookArguments can be reverted too.

Reply:
Updated accordingly.